packages/live-preview-sdk/src/inspectorMode/index.ts

```diff
@@ -0,0 +1,216 @@
+import { sendMessageToEditor } from '../helpers';
+import type { MessageFromEditor } from '../messages';
+import {
+  InspectorModeDataAttributes,
+  type InspectorModeChangedMessage,
+  InspectorModeEventMethods,
+} from './types';
+import { getAllTaggedElements, getInspectorModeAttributes } from './utils';
+
+export class InspectorMode {
+  private defaultLocale: string;
+  private targetOrigin: string[];
+
+  private isScrolling = false;
+  private scrollTimeout?: NodeJS.Timeout;
+
+  private isResizing = false;
+  private resizeTimeout?: NodeJS.Timeout;
+
+  private hoveredElement?: HTMLElement;
+  private taggedElements: Element[] = [];
+
+  constructor({ locale, targetOrigin }: { locale: string; targetOrigin: string[] }) {
+    this.defaultLocale = locale;
+    this.targetOrigin = targetOrigin;
+
+    this.bindHoverListener = this.bindHoverListener.bind(this);
+    this.bindScrollListener = this.bindScrollListener.bind(this);
+    this.bindMutationListener = this.bindMutationListener.bind(this);
+    this.bindResizeListener = this.bindResizeListener.bind(this);
+
+    this.handleTaggedElement = this.handleTaggedElement.bind(this);
+    this.sendAllElements = this.sendAllElements.bind(this);
+
+    // Attach interaction listeners
```

Comment:
nit: maybe it's just me being super picky but I think the naming is a little confusing here as these functions are already "bound", I do understand that it's a different type of binding just as a suggestion maybe something like

```
  this.handleHoverEvents();
  this.handleScrollEvents();
  this.handleMutationObserverEvents();
  this.handleResizeEvents();
```

then the handlers are bound to the class instance 🤷🏻‍♂️

Reply:
or even `this.addHoverEventListener...` etc.

Reply:
you want it, you get it :) 
instead of `bind` it's now `addXYZ`